contracts/DispenserInternal.sol

```diff
@@ -61,6 +94,11 @@ abstract contract DispenserInternal is IDispenserProvider, DealProvider, Dispens
         lockDealNFT.cloneVaultId(poolId, tokenPoolId);
     }
 
+    /// @notice Withdraws tokens from the provider if the withdrawable amount is greater than zero.
+    /// @dev Transfers the tokens from the owner to the `lockDealNFT` contract if available.
+    /// @param provider The simple provider from which tokens will be withdrawn.
+    /// @param poolId The unique identifier for the pool to withdraw from.
+    /// @param owner The address of the owner of the tokens being withdrawn.
     function _withdrawIfAvailable(
         ISimpleProvider provider,
```

Comment:
why do we need the provider here?

Reply:
yes, we can use `lockDealNFT.getWithdrawableAmount` instead of `provider.getWithdrawableAmount`

Reply:
let's do this in another PR